### ISSUE TITLE
pubsub: Unable to publish message from GAE Standard

### ISSUE THREAD

**Issue by d2g** [GoogleCloudPlatform/google-cloud-go#702]
Using GAE standard environment I am unable to publish messages. 

I created the following Repository (https://github.com/d2g/dg-pubsubtest) to demonstrate the issue and also tried utilising task queues to extend the timeout. In either case it hangs awaiting the `get(ctx)` response.

If I run this on the development server I get the following:
_Unsolicited response received on idle HTTP channel starting with "HTTP/1.1 408 Request Timeout\r\nContent-Length: 0\r\nContent-Type: text/plain\r\n\r\n"; err=<nil>_

Followed By
_ERROR: context expired before API call remote_socket/Resolve completed_

If I remove the `get(ctx)` the requests completes on the with a 200 but no message is delivered to the topic.

I apologise in advance if this is my issue, I did try the google-appengine-go forum first.

**Comment by zombiezen** [comment]
As per https://cloud.google.com/pubsub/docs/reference/libraries (first paragraph), we recommend using the older client library at [`google.golang.org/api/pubsub/v1`](https://godoc.org/google.golang.org/api/pubsub/v1) on App Engine Standard.

That said, publishing using this client should (mostly) work, as long as the Client is request-scoped.  Pull subscribing will probably not (especially synchronously in the same request), as the client is intended to be running as a long-running connection over the process, which App Engine Standard fundamentally doesn't support.  You can instead use [push subscriptions](https://cloud.google.com/pubsub/docs/push#app-engine-standard-endpoints).

**Comment by d2g** [comment]
I'll look into the older client to see if I can use that to workaround the issue.

In my example repository, the client is request-scoped and I tried to simplify the example as much as possible (only publishing a singular small message to an existing topic).

**Comment by broady** [comment]
Hi @d2g, for publishing, you can try this code:

https://gist.github.com/broady/c79a65cb49d7b6a56448b3345a23b3d6

**Comment by d2g** [comment]
@broady Thanks that works great. 

Is the intention to update the documentation to reflect that GAE standard only supports v1 (not that it's just recommended) or get the later client to provide some/minimal support?

**Comment by broady** [comment]
I think the current documentation is OK, but making the client better support GAE standard would also be good. That's up to @jba.

**Comment by jba** [comment]
Given the increased weight of `Topic.Publish`, it doesn't seem a good fit in a request-scoped setting. I think we shouldn't try to make this client work better on GAE standard.

**Comment by leighmcculloch** [comment]
I think it's worth updating the documentation [here](https://cloud.google.com/pubsub/docs/reference/libraries) that states that `recommend using the older Google APIs Client Libraries if running on Google App Engine standard environment`, to explicitly link to the library it is referencing: `google.golang.org/api/pubsub/v1`. It was tricky to figure out what this document meant by the "older Google APIs". It wasn't until I found this Github issue that I found out what I needed to import.
